inc/functions/net.php

```diff
@@ -3,16 +3,14 @@
 
 
 /**
+ * @param bool $trust_headers. If true, trust the `HTTP_X_FORWARDED_PROTO` header to check if the connection is HTTPS.
  * @return bool Returns if the client-server connection is an encrypted one (HTTPS).
  */
-function is_connection_secure(): bool {
-    if (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off') {
-        return true;
-    }
-    elseif (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] === 'https') {
-        return true;
-    }
-    else {
+function is_connection_secure(bool $trust_headers): bool {
```

Comment:
[Nit](https://graphite.dev/blog/what-are-nits): Unless there's a good reason to not access the configuration from this function, I would recommend not passing a `trust_headers` parameter but accessing configuration directly at line 12. Having this parameter puts on the caller the burden of doing-the-right-thing, which isn't obvious and sligthly error prone.

As a user of this function I'd just want to know if the connection is secure based on a good/sane definition of secure. Having to think/remember that I need to pass a parameter to make it do the right thing is a smell, imho. 😄

Reply:
I'd rather not make it dependant on global state more than strictly necessary. Such pattern has already bitten me in the arse when `openBoard()` gets called in the overboard building process.
On top of that, there isn't much burden on the caller here: it's just a single parameter that will always be casted to `bool` anyway.